source/fab/application.py

```diff
@@ -4,36 +4,50 @@
 # which you should have received as part of this distribution
 ##############################################################################
 from pathlib import Path
-from typing import Dict, Mapping, Type
+from typing import Dict, Type, Union, List
 
 from fab.database import SqliteStateDatabase
-from fab.language import Analyser
-from fab.language.fortran import FortranAnalyser, FortranWorkingState
-from fab.source_tree import ExtensionVisitor, FileInfoDatabase, TreeDescent
+from fab.language import Task, Command
+from fab.language.fortran import \
+    FortranAnalyser, \
+    FortranWorkingState, \
+    FortranPreProcessor
+from fab.source_tree import TreeDescent, ExtensionVisitor, FileInfoDatabase
 
 
 class Fab(object):
-    _extensions: Dict[str, Type[Analyser]] = {
-        '.F90': FortranAnalyser,
-        '.f90': FortranAnalyser
+    _extension_map: Dict[str, Union[Type[Task], Type[Command]]] = {
+        '.f90': FortranAnalyser,
+        '.F90': FortranPreProcessor,
     }
+    _command_flags_map: Dict[Type[Command], List[str]] = {}
```

Comment:
Havin `_command_flag` as a class variable and adding to it in the constructor means it will collect mappings from every instance. Is that the behaviour you want?

Reply:
I think the way the constructor adds to it means that it won't collect _duplicates_ at least, but beyond that I'd not really thought too much about it.  I'll move the initialisation into the constructor for now